GitUpKit/Core/GCFunctions.m

```diff
@@ -131,6 +135,27 @@ BOOL GCGitOIDFromSHA1Prefix(NSString* prefix, git_oid* oid, NSError** error) {
   return url.absoluteString;
 }
 
+int CGExchangeDataFromTempPathToPath(const char* tempPath, const char* path) {
+  int status;
+  struct statfs stat;
+  status = statfs(path, &stat);
+  if (status == 0) {
+    struct attrlist attrList = { ATTR_BIT_MAP_COUNT, 0, 0, ATTR_VOL_CAPABILITIES, 0, 0, 0 };
+    struct { u_int32_t length; vol_capabilities_attr_t attr; } attrBuf;
+    status = getattrlist(stat.f_mntonname, &attrList, &attrBuf, sizeof(attrBuf), 0);
+    if (status == 0) {
+      if ((attrBuf.attr.capabilities[VOL_CAPABILITIES_INTERFACES] & VOL_CAP_INT_EXCHANGEDATA) == VOL_CAP_INT_EXCHANGEDATA) {
+        status = exchangedata(tempPath, path, FSOPT_NOFOLLOW);
+      } else {
+        status = rename(tempPath, path);
+      }
+    }
+  } else {
+    status = rename(tempPath, path);
```

Comment:
Remove this statement: if `statfs` fails, then the function should fail immediately as this would be a fatal error